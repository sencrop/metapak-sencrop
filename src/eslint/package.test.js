'use strict';

const packageTransformer = require('./package');

describe('Main', () => {
  describe('Package transformer', () => {
    test('should work with an empty package.json', () => {
      expect(
        packageTransformer({
          metapak: {
            data: {
              files: 'lol.js',
              eslintConfigType: 'backend',
            },
          },
        }),
      ).toMatchSnapshot();
    });

    test('should work with an empty package.json for react', () => {
      expect(
        packageTransformer({
          metapak: {
            data: {
              files: 'lol.js',
              eslintConfigType: 'react',
            },
          },
        }),
      ).toMatchSnapshot();
    });
  });
});
