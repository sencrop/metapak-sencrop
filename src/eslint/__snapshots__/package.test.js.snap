// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Package transformer should work with an empty package.json 1`] = `
Object {
  "devDependencies": Object {
    "eslint": "^5.14.1",
    "eslint-plugin-import": "^2.16.0",
    "eslint-plugin-prettier": "^3.0.1",
    "lint-staged": "^3.0.1",
    "prettier": "^1.16.4",
  },
  "eslintConfig": Object {
    "env": Object {
      "es6": true,
      "jest": true,
      "mocha": true,
      "node": true,
    },
    "extends": "eslint:recommended",
    "parserOptions": Object {
      "ecmaVersion": 9,
      "sourceType": "module",
    },
    "plugins": Array [
      "prettier",
      "import",
    ],
    "rules": Object {
      "prettier/prettier": "error",
    },
  },
  "husky": Object {
    "hooks": Object {
      "pre-commit": "lint-staged",
    },
  },
  "lint-staged": Object {
    "*.{js,jsx}": "eslint",
  },
  "metapak": Object {
    "data": Object {
      "eslintConfigType": "backend",
      "files": "lol.js",
    },
  },
  "prettier": Object {
    "printWidth": 80,
    "proseWrap": "always",
    "semi": true,
    "singleQuote": true,
    "trailingComma": "all",
  },
  "scripts": Object {
    "lint": "eslint lol.js",
    "prettier": "prettier --write lol.js",
  },
}
`;

exports[`Main Package transformer should work with an empty package.json for create-react-app 1`] = `
Object {
  "devDependencies": Object {
    "eslint": "5.6.0",
    "eslint-plugin-import": "^2.16.0",
    "eslint-plugin-prettier": "^3.0.1",
    "lint-staged": "^3.0.1",
    "prettier": "^1.16.4",
  },
  "eslintConfig": Object {
    "extends": "react-app",
    "plugins": Array [
      "prettier",
    ],
    "rules": Object {
      "prettier/prettier": "error",
    },
  },
  "husky": Object {
    "hooks": Object {
      "pre-commit": "lint-staged",
    },
  },
  "lint-staged": Object {
    "*.{js,jsx}": "eslint",
  },
  "metapak": Object {
    "data": Object {
      "eslintConfigType": "create-react-app",
      "files": "lol.js",
    },
  },
  "prettier": Object {
    "printWidth": 80,
    "proseWrap": "always",
    "semi": true,
    "singleQuote": true,
    "trailingComma": "all",
  },
  "scripts": Object {
    "lint": "eslint lol.js",
    "prettier": "prettier --write lol.js",
  },
}
`;
